fms/models/hf/utils.py

```diff
@@ -101,3 +102,38 @@ def mask_2d_to_3d_bidirectional(
     )
 
     return mask_encoder.unsqueeze(1) == mask_decoder.unsqueeze(2)
+
+
+def wrap(model: nn.Module, **override_config_kwargs) -> "HFModelArchitecture":
+    """Wrap an FMS model, converting its API to one of and Huggingface model
+
+    Parameters
+    ----------
+    model: nn.Module
+        The FMS model to wrap (currently one of LLaMA or GPTBigCode)
+    override_config_kwargs
+        configuration parameters to override as a set of keyword arguments
+
+    Returns
+    -------
+    HFModelArchitecture
+        an HF adapted FMS model
+    """
+    from fms.models.hf.gpt_bigcode.modeling_gpt_bigcode_hf import (
+        HFAdaptedGPTBigCodeForCausalLM,
+    )
+    from fms.models.hf.llama.modeling_llama_hf import HFAdaptedLLaMAForCausalLM
+
+    register_fms_models()
+
+    __fms_to_hf_adapt_map = {
+        "LLaMA": HFAdaptedLLaMAForCausalLM,
+        "GPTBigCode": HFAdaptedGPTBigCodeForCausalLM,
```

Comment:
Maybe we could create a single global model list (with the class of the fms model, HF model, and HF config for each) then iterate through that in `register_fms_models` as well? I'm thinking it would be good to reduce/minimize the number of places that need to be updated for each HF-adapter we add.

Reply:
Added this to latest commit

Reply:
ideally, all mapping will be in a single place - a sort of catalog....